ev3dev.go

```diff
@@ -683,14 +683,14 @@ func stateFrom(data, _ string, err error) (MotorState, error) {
 	for _, s := range strings.Split(data, " ") {
 		bit, ok := motorStateTable[s]
 		if !ok {
-			return 0, fmt.Errorf("ev3dev: unrecognized motor state value: %s in [%s]", s, data)
+			return 0, newInvalidValueError(d, state, "unrecognized motor state", s, motorStates)
```

Comment:
This should probably do something like `keys(motorStateTable)` in place of `motorStates` where

```
func keys() []string {
    l := make([]string, 0, len(motorStateTable))
    for s := range motorStateTable {
        l = append(l, s)
    }
    sort.Strings(l)
    return l
}
```

because with the current API, it is possible for the user to mutate the `motorStates` slice.